src/controllers/lesson.js

```diff
@@ -0,0 +1,88 @@
+const getCategoriesOptions = require('~/utils/getCategoriesOption')
+const getMatchOptions = require('~/utils/getMatchOptions')
+const getSortOptions = require('~/utils/getSortOptions')
+const { lessonService } = require('~/services/lesson')
+
+class LessonController {
+  getLessons(req, res) {
+    const { title, sort, skip = 0, limit = 10, categories } = req.query
+    const userId = req.user.id
+    console.log(userId)
+    let categoriesOptions
+
+    if (Array.isArray(categories)) {
+      categoriesOptions = getCategoriesOptions(categories)
+    } else if (categories) {
+      categoriesOptions = [categories]
+    }
+
+    const match = getMatchOptions({
+      title,
+      categoryIDs: categoriesOptions
+    }) // Type Match = {title: string | string[], categoryIDs: string[]}
+
+    const sortOptions = getSortOptions(sort)
+    try {
+      const lessons = lessonService.getLessons(userId, match, sortOptions, +skip, +limit)
+      res.status(200).json(lessons)
+    } catch (e) {
+      console.error(e.message)
```

Comment:
The way I see it, you can add more details for error handlers. Something like this "Failed to process request', details: e.message"